back/app/services/user_custom_field_service.rb

```diff
@@ -1,10 +1,26 @@
 # frozen_string_literal: true
 
 class UserCustomFieldService
-  def delete_custom_field_values(field)
-    User
-      .where("custom_field_values \? '#{field.key}'")
-      .update_all("custom_field_values = custom_field_values - '#{field.key}'")
+  def delete_custom_field_values(field, current_user)
+    users = User.where("custom_field_values ? '#{field.key}'")
+    log = {}
+
+    users.each do |user|
+      log[user.id] = { field.key.to_s => user.custom_field_values[field.key] }
+    end
```

Comment:
We shouldn't be waiting for this (could iterate over thousands of users). Need to move this to perform_later Job.